### PR TITLE
Improve code in `testing.backend.BackendConfig`

### PR DESCRIPTION
**Pull request by niboshi**

### CODE REVIEW

File: chainer/testing/backend.py [hunk 1]
```diff
@@ -64,20 +61,16 @@ def _adjust_params(self):
...
     def device(self):
         if self._device is None:
             if self.use_cuda:
-                device = chainer.get_device(
-                    chainer.backends.cuda.Device(self.cuda_device))
+                device = backend.GpuDevice.from_device_id(self.cuda_device)
             elif self.use_chainerx:
-                device = chainer.get_device(self.chainerx_device)
+                device = backend.ChainerxDevice(
+                    chainerx.get_device(self.chainerx_device))
```
Comment (hvy) [inline chainer/testing/backend.py]: I prefer the previous simpler version 🤔

Comment (niboshi) [reply chainer/testing/backend.py]: Yes, and that makes the code safer.
For example, if `self.chainerx_device` was `1` for some reason, the previous code would not cause any error here.

Comment (hvy) [reply chainer/testing/backend.py]: Okay, thank you for the explanation. I think that the user input validation should take place before reaching this line if any but that's maybe another story.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Failures seems unrelated. LGTM!

**Pull request merged**